src/v/transform/rpc/client.cc

```diff
@@ -161,7 +160,8 @@ ss::future<cluster::errc> client::produce(
     produce_request req;
     req.topic_data.emplace_back(std::move(tp), std::move(batches));
     req.timeout = timeout;
-    co_return co_await do_produce_once(std::move(req));
+    co_return co_await retry(
```

Comment:
is produce idempotent? if not we risk having duplicates for example retrying on timeouts? Probably not an issue for now for atleast once semantics, just wanted to point it out.

Reply:
We do risk duplicates, but as you say because it's at least once semantics it's OK (and required actually to retry until success).